### PR TITLE
IDCOM-2102 is_authority controller change

### PR DESCRIPTION
**Pull request by dankelleher**
Update controller parameters for is_authority to support generative dids

### CODE REVIEW

File: sol-did/programs/sol-did/src/utils.rs [hunk 1]
```diff
@@ -1,19 +1,39 @@
 use crate::constants::DID_SOL_PREFIX;
+use crate::{id, DID_ACCOUNT_SEED};
+use anchor_lang::prelude::*;
+use lazy_static::lazy_static;
+use regex::Regex;
 use solana_program::keccak;
 use solana_program::secp256k1_recover::Secp256k1Pubkey;
 
+lazy_static! {
```
Comment (dankelleher) [inline sol-did/programs/sol-did/src/utils.rs:9]: One concern is this warning on build:

```
cargo-build-bpf child: cargo +bpf build --target bpfel-unknown-unknown --release
Error: Function _ZN5regex4exec11ExecBuilder5build17h83ceff945f02fa86E Stack offset of 7688 exceeded max offset of 4096 by 3592 bytes, please minimize large stack variables
```

Guessing this is caused by this block. Not sure what effect it has on compute . However I'm wondering if we should do this differenty, avoid using a regex by just using indexof/substring or similar string manipulation?

Comment (rado0x54) [reply sol-did/programs/sol-did/src/utils.rs:9]: hm, funny I used this `lazy_static` also before and ran into the same problem (when trying to return the `owner()` as a static Pubkey. Afterwards I decided against it.

File: sol-did/programs/sol-did/src/state/did_account.rs [hunk 2]
```diff
@@ -194,15 +257,21 @@ impl DidAccount {
 
     /// Returns true if `other` is a valid controller of this DID
     pub fn is_directly_controlled_by(&self, other: &DidAccount) -> bool {
-        other.other_controllers.contains(&self.to_string())
+        let this_did = normalize_did_cluster(self.to_string().as_str());
+        other
+            .other_controllers
```
Comment (rado0x54) [inline sol-did/programs/sol-did/src/state/did_account.rs:262]: I'm trying to understand. Why do we even need to support `other_controller`? All `did:sol` controllers are in `native_controllers`

File: sol-did/programs/sol-did/src/utils.rs (hunk 1 above)
Comment (dankelleher) [reply sol-did/programs/sol-did/src/utils.rs]: Removed after discussion

File: sol-did/programs/sol-did/src/state/did_account.rs (hunk 2 above)
Comment (dankelleher) [reply sol-did/programs/sol-did/src/state/did_account.rs]: fixed

**Pull request merged**